src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_LENGTH = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[ARRAY_LENGTH];
+        this.values = (V[]) new Object[ARRAY_LENGTH];
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size + 1; i++) {
+            if (key == null) {
+                if (keys[i] == null && values[i] != null) {
+                    size--;
+                }
+                break;
+            }
+            if (key.equals(keys[i])) {
+                size--;
+            }
```

Comment:
Interesting, but here is how it can be shortened :)
```suggestion
            if (key == keys[i] || key != null && key.equals(keys[i])) {
                values[i] = value;
                return;
            }
```